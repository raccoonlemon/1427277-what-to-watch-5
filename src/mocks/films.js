export const films = [
  {
    id: `1`,
    title: `The Grand Budapest Hotel`,
    genre: `drama`,
    year: 2014,
    director: `Wes Andreson`,
    duration: 124,
    starring:
    [
      `Bill Murray`,
      `Edward Norton`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`,
      `Tony Revoloru`,
      `Tom Wilkinson`,
      `Owen Wilkinson`,
      `Adrien Brody`,
      `Ralph Fiennes`,
      `Jeff Goldblum`
    ],
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
    Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    posterSmall: `img/the-grand-budapest-hotel-poster.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },

  {
    id: `2`,
    title: `Pulp fiction`,
    genre: `crime`,
    year: 1994,
    director: `Quentin Tarantino`,
    duration: 118,
    starring:
    [
      `Tim Roth`,
      `Amanda Plummer`,
      `Laura Lovelace`,
      `John Travolta`,
      `Samuel L. Jackson`,
      `Phil LaMarr`,
      `Frank Whaley`,
      `Burr Steers`,
      `Bruce Willis`
    ],
    description: `The lives of two mob hitmen, a boxer, a gangster and his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
    background: `img/pulp-fiction.jpg`,
    poster: `img/pulp-fiction.jpg`,
    posterSmall: `img/pulp-fiction.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },

  {
    id: `3`,
    title: `Moonrise Kingdom`,
    genre: `romance`,
    year: 2012,
    director: `Wes Anderson`,
    duration: 98,
    starring:
    [
      `Bruce Willis`,
      `Edward Norton`,
      `Bill Murray`,
      `Frances McDormand`,
      `Tilda Swinton`,
      `Jared Gilman`,
      `Kara Hayward`,
      `Jason Schwartzman`,
      `Bob Balaban`
    ],
    description: `A pair of young lovers flee their New England town, which causes a local search party to fan out to find them.`,
    background: `img/moonrise-kingdom.jpg`,
    poster: `img/moonrise-kingdom.jpg`,
    posterSmall: `img/moonrise-kingdom.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },

  {
    id: `4`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    genre: `fantasy`,
    year: 2018,
    director: `David Yates`,
    duration: 134,
    starring:
    [
      `Johnny Depp`,
      `Kevin Guthrie`,
      `Carmen Ejogo`,
      `Wolf Roth`,
      `Eddie Redmayne`,
      `Zoë Kravitz`,
      `Callum Turner`,
      `Derek Riddell`,
      `Cornell John`
    ],
    description: `The second installment of the "Fantastic Beasts" series featuring the adventures of Magizoologist Newt Scamander.`,
    background: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    posterSmall: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },

  {
    id: `5`,
    title: `Aviator`,
    genre: `biography`,
    year: 2004,
    director: `Martin Scorsese`,
    duration: 110,
    starring:
    [
      `Leonardo DiCaprio`,
      `Cate Blanchett`,
      `Kate Beckinsale`,
      `John C. Reilly`,
      `Alec Baldwin`,
      `Alan Alda`,
      `Jude Law`,
    ],
    description: `A biopic depicting the early years of legendary Director and aviator Howard Hughes' career from the late 1920s to the mid 1940s.`,
    background: `img/aviator.jpg`,
    poster: `img/aviator.jpg`,
    posterSmall: `img/aviator.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    id: `6`,
    title: `Shutter Island`,
    genre: `thriller `,
    year: 2010,
    director: `Martin Scorsese`,
    duration: 142,
    starring:
    [
      `Leonardo DiCaprio`,
      `Mark Ruffalo`,
      `Ben Kingsley`,
      `Michelle Williams`,
      `Emily Mortimer`,
      `Patricia Clarkson`,
      `Max von Sydow`,
    ],
    description: `In 1954, a U.S. Marshal investigates the disappearance of a murderer who escaped from a hospital for the criminally insane.`,
    background: `img/shutter-island.jpg`,
    poster: `img/shutter-island.jpg`,
    posterSmall: `img/shutter-island.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },

  {
    id: `7`,
    title: `Johnny English`,
    genre: `comedy `,
    year: 2003,
    director: `Peter Howitt`,
    duration: 109,
    starring:
    [
      `Rowan Atkinson`,
      `Natalie Imbruglia`,
      `Ben Miller`,
      `John Malkovich`,
      `Eddie Redmayne`,
      `Zoë Kravitz`,
      `Callum Turner`,
      `Derek Riddell`,
    ],
    description: `After a cyber-attack reveals the identity of all of the active undercover agents in Britain, Johnny English (Rowan Atkinson) is forced to come out of retirement to find the mastermind hacker.`,
    background: `img/johnny-english.jpg`,
    poster: `img/johnny-english.jpg`,
    posterSmall: `img/johnny-english.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },

  {
    id: `8`,
    title: `Seven Years in Tibet`,
    genre: `drama`,
    year: 1997,
    director: `Jean-Jacques Annaud`,
    duration: 125,
    starring:
    [
      `Brad Pitt`,
      `David Thewlis`,
      `BD Wong`,
      `Danny Denzongpa`,
      `Mako`,
      `Jamyang Jamtsho Wangchuk`,
      `Lhakpa Tsamchoe`,
      `Jetsun Pema`,
    ],
    description: `True story of Heinrich Harrer, an Austrian mountain climber who became friends with the Dalai Lama at the time of China's takeover of Tibet.`,
    background: `img/seven-years-in-tibet.jpg`,
    poster: `img/seven-years-in-tibet.jpg`,
    posterSmall: `img/seven-years-in-tibet.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
];

export const promoFilm = films[0];
