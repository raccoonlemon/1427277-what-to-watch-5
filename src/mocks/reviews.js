const texts = [
  `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
  `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight.`,
  `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
  `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
  `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`
];

export const reviews = [
  {
    id: 1,
    date: new Date(2020, 10, 10, 0, 0, 0, 0),
    author: `Ellie`,
    rating: 10,
    text: texts[0],
  },
  {
    id: 2,
    date: new Date(2020, 10, 12, 0, 0, 0, 0),
    author: `Max`,
    rating: 5,
    text: texts[1]
  },
  {
    id: 3,
    date: new Date(2020, 10, 12, 0, 0, 0, 0),
    author: `Parvati`,
    rating: 8,
    text: texts[2]
  },
  {
    id: 4,
    date: new Date(2020, 10, 12, 0, 0, 0, 0),
    author: `Felix`,
    rating: 8,
    text: texts[3]
  },
  {
    id: 5,
    date: new Date(2020, 10, 12, 0, 0, 0, 0),
    author: `Nyoka`,
    rating: 4,
    text: texts[4]
  }
];
