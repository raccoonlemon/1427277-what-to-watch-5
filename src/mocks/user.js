export const user = {
  id: `1`,
  email: `email@email.com`,
  name: `User Name`,
  avatarSrc: `src`,
};
