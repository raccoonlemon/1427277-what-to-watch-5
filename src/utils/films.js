import {ALL_GENRES_FILTER, FilmRating, FILM_RATING_SCALE} from "../const";

export const getFilmsByGenre = ({films, genreFilter}) =>{
  if (genreFilter === ALL_GENRES_FILTER) {
    return films.slice();
  }

  return films.filter((film)=>film.genre === genreFilter);
};

export const getGenresList = (films)=>{
  const MAX_GENRES_COUNT = 9;

  const genres = films.map((film)=>film.genre.toLowerCase());

  let uniqueGenres = [...new Set(genres)];

  uniqueGenres.sort();
  uniqueGenres = uniqueGenres.slice(0, MAX_GENRES_COUNT);
  uniqueGenres.unshift(ALL_GENRES_FILTER.toLowerCase());

  return uniqueGenres;
}


export const getSimilarFilms = (films, {genre, id}) =>{
  return films.filter((film)=>film.genre === genre && film.id !== id);
};

export const getAverageRating = (reviews)=>{
  const rewiewsCount = reviews.length;

  if (!reviews || !reviews.length) {
    return 0;
  }

  const totalRating = reviews.reduce((result, review)=>{
    return result + review.rating;
  }, 0);
  const averageRating = Math.round(totalRating / rewiewsCount * 10) / 10;

  return averageRating;
};


export const getRatingDescription = (reviews) =>{

  if (!reviews || !reviews.length) {
    return FilmRating.NOT_RATED;
  }
  const averageRating = getAverageRating(reviews);

  for (const filmRating of FILM_RATING_SCALE) {
    if (averageRating >= filmRating.minScore) {
      return filmRating.rating;
    }
  }

  return FilmRating.NOT_RATED;
};
