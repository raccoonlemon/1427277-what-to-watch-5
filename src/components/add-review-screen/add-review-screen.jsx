import PropTypes from 'prop-types';
import React, {useEffect, useState} from "react";
import {connect} from "react-redux";
import {Link} from "react-router-dom";
import {DEFAULT_RAITING_IN_REVIEW, MAX_RAITING_IN_REVIEW, ReviewPostErrorText, ReviewTextLength, UNKNOWN_ERROR} from "../../const";
import {useForm} from '../../hooks/useForm';
import {userRequested} from '../../store/actions/user';
import {fetchFilmById, postReview} from '../../store/api-actions';
import {selectFilm, selectIsFilmLoaded, selectIsReviewPostFailed, selectIsReviewPostRequested, selectUserRequestErrorCode} from "../../store/selectors";
import {filmShape} from "../../utils/props-validation";
import Header from "../header/header";

const ratingItems = [];
for (let index = 1; index <= MAX_RAITING_IN_REVIEW; index++) {
  ratingItems.push({
    id: `star-${index}`,
    value: index,
    title: `Rating ${index}`
  });
}

const validate = ({reviewText, rating})=>{
  let isValid = true;
  const messages = [];

  if (reviewText.length < ReviewTextLength.MIN) {
    isValid = false;
    messages.push(`Minimun review length - ${ReviewTextLength.MIN} symbols. Current lenght - ${reviewText.length}.`);
  }

  if (reviewText.length > ReviewTextLength.MAX) {
    isValid = false;
    messages.push(`Maximum review length - ${ReviewTextLength.MAX} symbols. Need to delete - ${reviewText.length - ReviewTextLength.MAX}.`);
  }

  if (!rating) {
    isValid = false;
    messages.push(`Please, select the rating score.`);
  }

  return {isValid, messages};
};

export const AddReviewScreen = (props) => {
  const {film, isFilmLoaded, loadFilmAction, postReviewAction, id, isRequested, isRequestFailed, errorCode} = props;
  const {title, poster, background} = film;

  useEffect(() => {
    if (!isFilmLoaded) {
      loadFilmAction(id);
    }
  });

  const initialValues = {reviewText: ``, rating: DEFAULT_RAITING_IN_REVIEW};
  const {validation, values, changeValue} = useForm(initialValues, validate);
  const {reviewText, rating} = values;

  const [isSubmitButtonActive, setIsSubmitButtonActive] = useState(false);

  useEffect(()=>{
    setIsSubmitButtonActive(validation.isValid);
  }, [validation, isRequested]);

  const filmScreenLink = `/films/${id}`;
  return (
    <React.Fragment>
      <section className="movie-card movie-card--full">
        <div className="movie-card__header">
          <div className="movie-card__bg">
            <img src={background} alt={title} />
          </div>

          <h1 className="visually-hidden">WTW</h1>

          <Header>
            <nav className="breadcrumbs">
              <ul className="breadcrumbs__list">
                <li className="breadcrumbs__item">
                  <Link className="breadcrumbs__link" to={filmScreenLink}>{title}</Link>
                </li>
                <li className="breadcrumbs__item">
                  <a className="breadcrumbs__link">Add review</a>
                </li>
              </ul>
            </nav>
          </Header>

          <div className="movie-card__poster movie-card__poster--small">
            <img src={poster} alt={title} width="218" height="327" />
          </div>
        </div>

        <div className="add-review">
          <form action="#" className="add-review__form" onSubmit = {()=>{}}>
            <div className="rating">
              <div className="rating__stars">
                {ratingItems.map((item)=>(
                  <React.Fragment key={item.id}>
                    <input className="rating__input"
                      id={item.id}
                      type="radio"
                      name="rating"
                      value={item.value}
                      defaultChecked ={item.value === rating}
                      onChange = {({target})=>
                        changeValue({rating: parseInt(target.value, 10)})}/>
                    <label className="rating__label" htmlFor={item.id}>{item.title}</label>
                  </React.Fragment>
                ))}
              </div>
            </div>

            <div className="add-review__text">
              <textarea
                className="add-review__textarea"
                name="review-text"
                id="review-text"
                placeholder="Review text"
                onChange = {({target})=>changeValue({reviewText: target.value})}
                value={reviewText}>
              </textarea>
              <div className="add-review__submit">
                <button
                  className="add-review__btn"
                  type="submit"
                  disabled={!isSubmitButtonActive}
                  onClick={(evt)=>{
                    evt.preventDefault();
                    postReviewAction(id, rating, reviewText);
                  }}>Post</button>
              </div>
            </div>
            {isRequestFailed && <div>{ReviewPostErrorText[errorCode] || UNKNOWN_ERROR}</div>}
            {validation.messages.map((item, index)=><div key={index}>{item}</div>)}

          </form>
        </div>

      </section>
    </React.Fragment>
  );
};

AddReviewScreen.propTypes = {
  id: PropTypes.string.isRequired,
  film: filmShape.isRequired,
  isRequested: PropTypes.bool.isRequired,
  isRequestFailed: PropTypes.bool.isRequired,
  errorCode: PropTypes.number.isRequired,
  isFilmLoaded: PropTypes.bool.isRequired,
  loadFilmAction: PropTypes.func.isRequired,
  postReviewAction: PropTypes.func.isRequired,
};

const mapStateToProps = (state, {id}) => (
  {
    film: selectFilm(state),
    isFilmLoaded: selectIsFilmLoaded(id)(state),
    isRequested: selectIsReviewPostRequested(state),
    isRequestFailed: selectIsReviewPostFailed(state),
    errorCode: selectUserRequestErrorCode(state),
  });

const mapDispatchToProps = (dispatch) => ({
  loadFilmAction(id) {
    dispatch(fetchFilmById(id));
  },
  postReviewAction(id, rating, comment) {
    dispatch(userRequested());
    dispatch(postReview(id, rating, comment));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(AddReviewScreen);
