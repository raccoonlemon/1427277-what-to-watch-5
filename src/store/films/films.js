import {createSelector} from "reselect";
import {MAX_SIMILAR_FILM_COUNT} from "../../const";
import {extend} from "../../utils/common";
import {getSimilarFilms} from "../../utils/films";
import {NameSpace} from "../namespace";

export const ActionType = {
  LOAD_FILMS: `LOAD_FILMS`,
  LOAD_FILM: `LOAD_FILM`,
  LOAD_PROMO_FILM: `LOAD_PROMO_FILM`,
  LOAD_FAVORITE_FILMS: `LOAD_FAVORITE_FILMS`,
  UPDATE_IS_MOVIE_FAVOTIRE: `UPDATE_IS_MOVIE_FAVOTIRE`,
  UPDATE_FILMS_INFO: `UPDATE_FILMS_INFO`,
};

const initialState = {
  films: [],
  reviews: [],
  promoFilm: {},
  film: {},
  filteredFilms: [],
  favoriteFilms: []
};

// Reducer

export const filmsReducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.LOAD_FILMS:
      return extend(state, {films: action.payload});

    case ActionType.LOAD_FILM:
      return extend(state, {film: action.payload});

    case ActionType.LOAD_PROMO_FILM:
      return extend(state, {promoFilm: action.payload});

    case ActionType.LOAD_FAVORITE_FILMS:
      return extend(state, {favoriteFilms: action.payload});

    case ActionType.UPDATE_FILMS_INFO:
      return extend(state, {
        film: action.payload,
        films: state.films.map((item)=>action.payload.id === item.id ? action.payload : item),
        promoFilm: state.promoFilm.id === action.payload.id ? action.payload : state.promoFilm,
        favoriteFilms: action.payload.isFavorite
          ? state.favoriteFilms.concat([action.payload])
          : state.favoriteFilms.filter((item)=> item.id !== action.payload.id)
      });
  }

  return state;
};

// Actions

export const loadFilms = (films) => ({
  type: ActionType.LOAD_FILMS,
  payload: films,
});

export const loadFilm = (film) => ({
  type: ActionType.LOAD_FILM,
  payload: film,
});

export const loadPromoFilm = (film) => ({
  type: ActionType.LOAD_PROMO_FILM,
  payload: film,
});

export const loadFavoriteFilms = (films) => ({
  type: ActionType.LOAD_FAVORITE_FILMS,
  payload: films,
});

export const updateFilmsInfo = (film)=>({
  type: ActionType.UPDATE_FILMS_INFO,
  payload: film,
});


// Selectors

const nameSpace = NameSpace.FILMS;

export const selectFilms = (state) => state[nameSpace].films;
export const selectFavoriteFilms = (state) => state[nameSpace].favoriteFilms;
export const selectFilm = (state) => state[nameSpace].film;
export const selectPromoFilm = (state) => state[nameSpace].promoFilm;
export const selectIsFilmLoaded = (id) => (state) => selectFilm(state).id === id;

export const selectSimilarFilms = createSelector(
    [selectFilms, selectFilm],
    (films, film)=> getSimilarFilms(films, film).slice(0, MAX_SIMILAR_FILM_COUNT)
);

