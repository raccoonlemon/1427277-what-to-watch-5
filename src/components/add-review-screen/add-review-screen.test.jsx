import React from "react";
// import {MemoryRouter} from "react-router-dom";
import ShallowRenderer from "react-test-renderer/shallow";
import {film} from "../../mocks/films";
import {AddReviewScreen} from "./add-review-screen";

const noop = () => {};
const id = `1`;
const mockFilm = film;
const renderer = new ShallowRenderer();

it(`<AddReviewScreen> renders correctly`, () => {
  renderer.render(
      <AddReviewScreen
        id ={id}
        film = {mockFilm}
        isFilmLoaded = {true}
        isRequested = {false}
        isRequestFailed = {false}
        errorCode = {0}
        loadFilmAction={noop}
        postReviewAction={noop}/>
  );
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});
