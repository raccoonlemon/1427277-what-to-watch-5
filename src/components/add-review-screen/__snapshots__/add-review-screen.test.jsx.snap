// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddReviewScreen> renders correctly 1`] = `
<React.Fragment>
  <section
    className="movie-card movie-card--full"
  >
    <div
      className="movie-card__header"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="The Grand Budapest Hotel"
          src="img/bg-the-grand-budapest-hotel.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <Header
        className=""
        needToShowUserMenu={true}
      >
        <nav
          className="breadcrumbs"
        >
          <ul
            className="breadcrumbs__list"
          >
            <li
              className="breadcrumbs__item"
            >
              <Link
                className="breadcrumbs__link"
                to="/films/1"
              >
                The Grand Budapest Hotel
              </Link>
            </li>
            <li
              className="breadcrumbs__item"
            >
              <a
                className="breadcrumbs__link"
              >
                Add review
              </a>
            </li>
          </ul>
        </nav>
      </Header>
      <div
        className="movie-card__poster movie-card__poster--small"
      >
        <img
          alt="The Grand Budapest Hotel"
          height="327"
          src="img/the-grand-budapest-hotel-poster.jpg"
          width="218"
        />
      </div>
    </div>
    <div
      className="add-review"
    >
      <form
        action="#"
        className="add-review__form"
        onSubmit={[Function]}
      >
        <div
          className="rating"
        >
          <div
            className="rating__stars"
          >
            <React.Fragment>
              <input
                className="rating__input"
                defaultChecked={false}
                id="star-1"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={1}
              />
              <label
                className="rating__label"
                htmlFor="star-1"
              >
                Rating 1
              </label>
            </React.Fragment>
            <React.Fragment>
              <input
                className="rating__input"
                defaultChecked={false}
                id="star-2"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={2}
              />
              <label
                className="rating__label"
                htmlFor="star-2"
              >
                Rating 2
              </label>
            </React.Fragment>
            <React.Fragment>
              <input
                className="rating__input"
                defaultChecked={true}
                id="star-3"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={3}
              />
              <label
                className="rating__label"
                htmlFor="star-3"
              >
                Rating 3
              </label>
            </React.Fragment>
            <React.Fragment>
              <input
                className="rating__input"
                defaultChecked={false}
                id="star-4"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={4}
              />
              <label
                className="rating__label"
                htmlFor="star-4"
              >
                Rating 4
              </label>
            </React.Fragment>
            <React.Fragment>
              <input
                className="rating__input"
                defaultChecked={false}
                id="star-5"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={5}
              />
              <label
                className="rating__label"
                htmlFor="star-5"
              >
                Rating 5
              </label>
            </React.Fragment>
          </div>
        </div>
        <div
          className="add-review__text"
        >
          <textarea
            className="add-review__textarea"
            id="review-text"
            name="review-text"
            onChange={[Function]}
            placeholder="Review text"
            value=""
          />
          <div
            className="add-review__submit"
          >
            <button
              className="add-review__btn"
              disabled={true}
              onClick={[Function]}
              type="submit"
            >
              Post
            </button>
          </div>
        </div>
      </form>
    </div>
  </section>
</React.Fragment>
`;
